docker/scout/scripts/main.py

```diff
@@ -619,12 +621,22 @@ def update_bpt_gauge(bpt_gauge: Gauge, bpt_name: str, bpt_address: str) -> None:
             bpt_address,
             "mcap"
         ).set(bpt_cummulative_price)
+
+        usd_prices_by_token_address[token_address] = bpt_cummulative_price / (bpt_total_supply / 10 ** bpt_contract.decimals())
```

Comment:
When I was doing work for daowatch I was hesitant to update this dict. This is a global object modified by too many functions now and it's VERY hard to debug it if something goes wrong with prices.

Maybe OK for now, but I would mark it deprecated and rethink entire pricing collection

Reply:
Yeah it sounds good to me.  I wasn't sure how to otherwise handle it.  I just pop pricing in when available and then pull it back out.  In the case here, we are using the data right after we put it in.  I agree that it's not well tested/easy to verify and that this probably needs to be rethought.  I do not have the skillz to do so tho.